File: app/src/main/java/com/example/miniproject/Database/Dao/SourceDao.java
package com.example.miniproject.Database.Dao;


import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.example.miniproject.Model.SourceModel;

import java.util.List;

@Dao
public interface SourceDao {
    @Query("SELECT * FROM tblSource")
    LiveData<List<SourceModel>> loadAllSources();

    @Query("SELECT * FROM tblSource WHERE id = :id")
    LiveData<SourceModel> getSourceById(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllSources(List<SourceModel> sourceModels);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateSource(SourceModel SourceModel);

    @Delete
    void deleteSource(SourceModel sourceModel);

    @Query("DELETE FROM tblSource WHERE createdDateTime < :time")
    void deleteAllSources(long time);
}

File: app/src/main/java/com/example/miniproject/database/dao/SourceDao.kt
package com.example.miniproject.database.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.miniproject.model.SourceModel

@Dao
interface SourceDao {
    @Query("SELECT * FROM tblSource")
    fun loadAllSources(): LiveData<List<SourceModel>?>?

    @Query("SELECT * FROM tblSource WHERE id = :id")
    fun getSourceById(id: String?): LiveData<SourceModel?>?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllSources(sourceModels: List<SourceModel?>?)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateSource(SourceModel: SourceModel?)

    @Delete
    fun deleteSource(sourceModel: SourceModel?)

    @Query("DELETE FROM tblSource WHERE createdDateTime < :time")
    fun deleteAllSources(time: Long)
}